### DISCUSSION TITLE
Unexpected behavior when dealing with incorrect Exif tag with far future date

### DISCUSSION THREAD

**Discussion by aaronrudkin** [photoprism/photoprism#5364] [Q&A]
#### 1. What is not working as documented?

I imported a file whose Exif date is incorrectly 12/31/2050. The file, when imported to Photoprism, reads as having 12/31/2023. I assume what happens is that Photoprism somewhat reasonably calculates that 2050 is an impossible year, and sets any date > the current year to the current year. The fact that the resulting month and day remains impossible is not sanitized.

#### 2. How can we reproduce it?

Import a photo with a far future exif original capture time

#### 3. What behavior do you expect?

Either (a) do not sanitize Exif data and allow me to import incorrect data asis or (b) discard the Exif capture time data entirely if it is invalid. Both of these would be preferable than importing the file as being taken in the near future. This makes it somewhat more difficult to troubleshoot and correct. Basically the bug here is just from a UX perspective, the files in Photoprism have a different metadata error than the equivalent original files outside Photoprism, making it harder to diagnose or correct the issues.

#### 4. What could be the cause of your problem?

As above, this appears to be a sanitizing issue with Exif data on the year of the capture time field.

#### 5. Can you provide us with example files for testing, error logs, or screenshots?

This is an example file. The original file was taken in 2010 on a point-and-shoot digital camera. Obviously I am aware that the 12/31/2050 date is incorrect.

![DSCF5356](https://github.com/photoprism/photoprism/assets/14701267/4ed48f18-a781-4700-a487-6e87c331120b)

#### 6. Which software versions do you use?

Photoprism on Synology NAS (running Synology 6.x, run through Docker), current buid, SQLite database. I am an Essentials member, though again I suspect this is all OS independent.

#### 7. On what kind of device is PhotoPrism installed?

Synology NAS, 20GB RAM; none of this seems hardware dependent.

#### 8. Do you use a Reverse Proxy, Firewall, VPN, or CDN?

No.

**Comment by lastzero** [comment]
It's not a bug as it works as intended. If we need to document this better, feel free to help with this as we are currently out of the office.

I guess for every solution we could implement, there will be one user who thinks it should be different. The logic might be further improved, but it's already pretty good and prevents setting the year 2050 which is completely impossible. Doesn't mean it prevents ALL dates that are impossible, just the most severe glitches.

**Comment by lastzero** [comment]
If you check other (closed) issues related to this, you should be able to find extensive discussions on this including opinions from a number of users.
